src/mappings/referrer.ts

```diff
@@ -17,54 +19,69 @@ export function handleReferralCodeCreated(event: OnReferralCodeCreated): void {
 }
 
 export function handleReferralCodeUpserted(event: OnReferralCodeUpserted): void {
-    let existingReferralCode = getReferralCode(event.params.oldReferralCode)
-    let trader = getOrCreateTrader(event.params.addr)
-    // Only proceed any action, if the referee has a code attached
-    if (existingReferralCode) {
-        // Referee removes code
-        if (event.params.action == 1) {
-            // trader no longer has a referee code
-            trader.refereeCode = null
-            trader.save()
-
-            // update referee count for code
-            existingReferralCode.numReferees = existingReferralCode.numReferees.minus(BI_ONE)
-            existingReferralCode.save()
-            return
-        }
+    if (!event.params.oldReferralCode) {
+        return
+    }
 
-        // add and update should only be possible if newReferralCode contains a value
-        if (event.params.newReferralCode != "") {
-            let newReferralCode = getReferralCode(event.params.newReferralCode)
-            let newReferralCodeDayData = getReferralCodeDayData(event, newReferralCode.id)
-            if (event.params.action == 2) {
-                // nothing changes here
-                if (event.params.oldReferralCode == event.params.newReferralCode) return
-            }
+    const oldRefCode = getReferralCode(event.params.oldReferralCode)
+    if (!oldRefCode) {
+        return
+    }
 
-            // uptick new referee daydata when an update or new referee is added
-            let newReferralReferees = newReferralCodeDayData.newReferees
-            newReferralReferees.push(event.params.addr.toHexString())
-            newReferralCodeDayData.newReferees = newReferralReferees
-            newReferralCodeDayData.save()
+    const ACTION_REMOVE_CODE = 1
+    const ACTION_REPLACE_CODE = 2
```

Comment:
According to the contract source code, I think we missed the case `ACTION_CREATE_CODE = 0` when user filled in his referral code for the first time.

Originally this `ADD` case is handled from line 38.

Reply:
got it. added a new switch case to handle `ADD` case